src/api/status/public/js/build-log/datetime-calculator.js

```diff
@@ -0,0 +1,45 @@
+const buildDateCaltulator = document.getElementById('calculatorData');
+const buildLoading = document.getElementById('build-loading');
+const buildSuccess = document.getElementById('build-success');
+const buildFail = document.getElementById('build-fail');
+
+export default function calculatorBuildInfo(data) {
+  const startedAtValue = data.startedAt;
+  const stoppedAtValue = data.stoppedAt;
+  const resultValue = data.result;
+
+  if (startedAtValue !== 'null' && stoppedAtValue === 'null') {
+    const date = Math.abs((new Date(startedAtValue).getTime() / 1000).toFixed(0));
+    const currentDate = Math.abs((new Date().getTime() / 1000).toFixed(0));
+    const diff = currentDate - date;
+    const hours = Math.floor(diff / 3600) % 24;
+    const minutes = Math.floor(diff / 60) % 60;
+    const seconds = diff % 60;
+    buildDateCaltulator.innerText = `${hours} h ${minutes} m ${seconds} s`;
+    buildFail.hidden = true;
+    buildSuccess.hidden = true;
+    buildLoading.hidden = false;
```

Comment:
It's easier to have only one icon placeholder in the HTML and use JavaScript to add icon in, rather than managing the display of all three icons.

```js
const buildIcon = document.getElementById("#build-icon");
buildIcon.innerHTML = "motion_photos_on"
buildIcon.classList.add("fa-spin");
```